### PR TITLE
[AKS] `az aks`: Add `--gpu-instance-profile` for Nvidia multi-instan…

### PR DESCRIPTION
**Pull request by alexeldeib**
…ce gpu

**Related command**

az aks create --gpu-instance-profile 
az aks nodepool add --gpu-instance-profile

**Description**

Support multi instance gpus for AKS. 

**Testing Guide**

see live test 

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_agentpool_decorator.py [hunk 1]
```diff
@@ -1267,6 +1267,26 @@ def common_get_no_wait(self):
         )
         self.assertEqual(ctx_1.get_no_wait(), False)
 
+    def common_set_up_gpu_propertes(self):
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_agentpool_decorator.py:1270]: missing several tests like [common_get_gpu_instance_profile](https://github.com/Azure/azure-cli-extensions/blob/main/src/aks-preview/azext_aks_preview/tests/latest/test_agentpool_decorator.py#L169), [standalone_test_get_gpu_instance_profile](https://github.com/Azure/azure-cli-extensions/blob/main/src/aks-preview/azext_aks_preview/tests/latest/test_agentpool_decorator.py#L283), [mc_test_get_gpu_instance_profile](https://github.com/Azure/azure-cli-extensions/blob/main/src/aks-preview/azext_aks_preview/tests/latest/test_agentpool_decorator.py#L318), [standalone_test_set_up_gpu_propertes](https://github.com/Azure/azure-cli-extensions/blob/main/src/aks-preview/azext_aks_preview/tests/latest/test_agentpool_decorator.py#L471), [mc_test_set_up_gpu_propertes](https://github.com/Azure/azure-cli-extensions/blob/main/src/aks-preview/azext_aks_preview/tests/latest/test_agentpool_decorator.py#L563)

Comment (alexeldeib) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_agentpool_decorator.py:1270]: oops, will add

Comment (alexeldeib) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_agentpool_decorator.py:1270]: think I got them all?

Comment (FumingZhang) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_agentpool_decorator.py:1270]: Yes, but this function seems to be called nowhere, maybe remove it?

Queued a [pipeline](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=58959702&view=results) to validate the live test case.

Comment (alexeldeib) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_agentpool_decorator.py]: seems good?

Comment (FumingZhang) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_agentpool_decorator.py]: Oops, sorry for the late reply. Looks all good to me :shipit:

**Review by FumingZhang — APPROVED** [review]
LGTM

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**